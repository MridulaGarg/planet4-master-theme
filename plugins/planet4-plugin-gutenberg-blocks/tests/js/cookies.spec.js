import {
  createNewPost,
  enablePageDialogAccept,
  getEditedPostContent,
  insertBlock,
} from '@wordpress/e2e-test-utils';

import {
  selectBlockByName,
  typeInInputWithPlaceholderLabel,
  typeInInputWithCheckbox,
} from './e2e-tests-helpers';

import 'expect-puppeteer';

describe( 'Cookies block', () => {
  beforeAll( async () => {
    // This helps in overriding Wordpress dialogs preventing actions
    await enablePageDialogAccept();
  } );
  beforeEach( async () => {
    // Before running each test, go to the create post page
    await createNewPost( {
      postType: 'page',
      title: 'Test Cookies block',
    } );

    // Insert block by title
    await insertBlock( 'Cookies' );
  } );

  // This is the first test, tests starts with it().
  it( 'is inserted into the Editor', async () => {
    // Check if block was inserted
    expect( await page.$( '[data-type="planet4-blocks/cookies"]' ) ).not.toBeNull();

    expect( await getEditedPostContent() ).toMatchSnapshot();
  }, 25000);

  it( 'should work with all inputs', async () => {
    await selectBlockByName( 'planet4-blocks/cookies' );

    // Add richtext inputs.
    await typeInInputWithPlaceholderLabel( 'Enter title', 'Cookies block' );
    await typeInInputWithPlaceholderLabel( 'Enter description', 'Test description' );

    await typeInInputWithCheckbox( 'necessary_cookies', 'Necessary cookies' );
    await typeInInputWithPlaceholderLabel( 'Enter necessary cookies description', 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.' );

    await typeInInputWithCheckbox( 'all_cookies', 'Other Cookies' );
    await typeInInputWithPlaceholderLabel( 'Enter all cookies description', 'Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.' );

    expect( await getEditedPostContent() ).toMatchSnapshot();
  });
} );
