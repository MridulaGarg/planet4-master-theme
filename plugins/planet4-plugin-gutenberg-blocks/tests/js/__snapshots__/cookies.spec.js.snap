// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookies block is inserted into the Editor 1`] = `"<!-- wp:planet4-blocks/cookies /-->"`;

exports[`Cookies block should work with all inputs 1`] = `"<!-- wp:planet4-blocks/cookies {\\"title\\":\\"Cookies block\\",\\"description\\":\\"Test description\\",\\"necessary_cookies_name\\":\\"Necessary cookies\\",\\"necessary_cookies_description\\":\\"Lorem Ipsum is simply dummy text of the printing and typesetting industry.\\",\\"all_cookies_name\\":\\"Other Cookies\\",\\"all_cookies_description\\":\\"Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.\\"} /-->"`;
